### PR TITLE
Add pagination example to Client4

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
A community member asked about how to use pagination. Our go docs are lacking an example, so this PR adds one. We can link for another future question about pagination.

#### Ticket Link
NONE

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/public/model/client4_test.go [hunk 1]
```diff
@@ -1,17 +1,21 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-package model
+package model_test
```
Comment (hanzei) [inline server/public/model/client4_test.go:4]: That allows the reader to copy the code

File: server/public/model/client4_test.go [hunk 2]
```diff
@@ -149,3 +153,30 @@ func TestClient4RequestCancellation(t *testing.T) {
...
+	for {
+		users, _, err := client.GetUsers(context.TODO(), page, perPage, "")
+		if err != nil {
+			log.Printf("error fetching users: %v", err)
+			return
+		}
+
+		if len(users) == 0 {
+			break
+		}
```
Comment (streamer45) [inline server/public/model/client4_test.go:173]: I generally tend to break as soon as the length is less than the page size. That can prevent making an additional call and potentially getting an error (depending on how the call is implemented that is).

File: server/public/model/client4_test.go [hunk 3]
```diff
@@ -149,3 +153,30 @@ func TestClient4RequestCancellation(t *testing.T) {
...
+
+		allUsers = append(allUsers, users...)
```
Comment (streamer45) [inline server/public/model/client4_test.go:175]: I sort of understand why we are doing this in the context of the example but in most cases I'd consider this to be an anti-pattern, having to paginate only to then have all the data in a single object. I'd expect this loop to print users as it's getting them.

**Review by streamer45 — COMMENTED** [review]
Nice addition! Left a couple of comments, nothing blocking but some potential suggestions.

File: server/public/model/client4_test.go (hunk 2 above)
Comment (hanzei) [reply server/public/model/client4_test.go]: Good call :+1:

File: server/public/model/client4_test.go (hunk 3 above)
Comment (hanzei) [reply server/public/model/client4_test.go]: Good point. I agree that it's an anti-pattern but a commonly used one.

Anyway, we don't need to promote it. Moved the print into the loop.

**Review by streamer45 — APPROVED** [review]
Thanks @hanzei , looks great :+1:

**Pull request merged**